pkg/utils/apiserver.go

```diff
@@ -0,0 +1,35 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetClusterAPIServerInfo returns the cluster API server URL
+// In Hypershift API Server and internal API Server are not required to be in a subdomain of the base domain.
+//
+// For example, given the base domain `openshift.example.com`,
+// an API server could be in `api.hypershift.local`.
+//
+// It uses this API https://docs.openshift.com/container-platform/4.7/rest_api/config_apis/dns-config-openshift-io-v1.html
+func GetClusterAPIServerInfo(ctx context.Context, configClient configv1client.ConfigV1Interface) ([]string, error) {
+	infra, err := configClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return []string{}, err
+	}
+	if infra == nil {
```

Comment:
The idea below was to check `infra.Status == nil` to prevent possible panic.

Reply:
I know, but through the unit tests I couldn't set it as nil because of a golang failure. Do you think it's possible for the struct to be initialized as `nil`, even if golang causes an error, as to reach there that way?

Reply:
Oh OK so the Status attribute is not a pointer so it can't be nil. Sorry for the noise.

Reply:
No problem. Actually, `infra` object could be nil, so I caught that possibility. Thanks for the revision.